### PR TITLE
Allow to add custom handlers in cartridge role

### LINKED ISSUE #120: Support custom handlers in metrics role
https://github.com/tarantool/metrics/blob/a780a21484b481c2e4318b0288282e744a4b8a98/cartridge/roles/metrics.lua#L6

### PR DESCRIPTION
**Pull request by yngvar-antonsson**
I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation (README and rst)

Close #120, #242

### CODE REVIEW

File: cartridge/roles/metrics.lua [hunk 1]
```diff
@@ -71,7 +73,7 @@ local function validate_routes(export)
     for _, v in ipairs(export) do
         v.path = remove_side_slashes(v.path)
         assert(type(v.path) == 'string', 'export.path must be string')
-        assert(handlers[v.format], 'format must be "json", "prometheus" or "health"')
+        assert(metrics_vars.handlers[v.format], 'format must be in handlers list')
```
Comment (DifferentialOrange) [inline cartridge/roles/metrics.lua:76]: Maybe we should keep on default handlers list in error message? I think this one may confuse users.

File: test/integration/cartridge_role_test.lua [hunk 2]
```diff
@@ -175,7 +175,7 @@ g.test_validate_config_invalid_export_format = function()
                 },
             },
         }
-    }, 'format must be "json", "prometheus" or "health"')
+    }, 'format must be  "json", "prometheus" or "health", or in custom handlers list')
```
Comment (rosik) [inline test/integration/cartridge_role_test.lua:178]: Double space?

File: doc/monitoring/getting_started.rst [hunk 3]
```diff
@@ -218,3 +218,15 @@ via configuration.
 
        local metrics = require('cartridge.roles.metrics')
        metrics.set_labels({ ['my-custom-label'] = 'label-value'} )
+
+#. To set custom handlers, you may use the ``set_custom_handlers`` function.
+   You may use it only once, in your ``init.lua`` file.
```
Comment (rosik) [inline doc/monitoring/getting_started.rst:223]: Why only once?

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `tnt_clock_delta` metric to compute clock difference on instances
 - set custom global labels in config and with `set_labels` function [#259](https://github.com/tarantool/metrics/issues/259)
+- set custom handlers with `set_custom_handlers` function [#120](https://github.com/tarantool/metrics/issues/120),
+  [#242](https://github.com/tarantool/metrics/issues/242)
```
Comment (rosik) [inline CHANGELOG.md:23]: How's #242 related to this patch?

Comment (yngvar-antonsson) [reply CHANGELOG.md:23]: #120 and #242 is about a same thing - custom handlers. Users could customize health response with custom health handler

**Comment by yngvar-antonsson** [comment]
Closed because of https://github.com/tarantool/metrics/issues/242#issuecomment-890057341

**Pull request closed without merging**